test/decl/ext/objc_implementation_direct_to_storage.swift

```diff
@@ -4,22 +4,20 @@
 
 import objc_implementation_internal
 
+// FIXME: [availability] An implementation that is less available than the interface it implements should be diagnosed
 @available(*, unavailable)
 @objc @implementation extension ObjCPropertyTest {
-  // FIXME: Shouldn't this be on the `@available` above?
-  // expected-note@+1 {{'prop1' has been explicitly marked unavailable here}}
   let prop1: Int32
 
-  // expected-note@+1 2 {{'prop2' has been explicitly marked unavailable here}}
   var prop2: Int32 {
     didSet {
-      _ = prop2 // expected-error {{'prop2' is unavailable}}
+      _ = prop2
```

Comment:
@beccadax I wanted to give you a heads up about the changes I'm making to this test since you just added it. The behavior that the test was reproducing before was incorrect, since the context of the methods in the `@objc @implementation extension` are themselves as unavailable as the properties and therefore the diagnostic should be supressed. After the refactoring in this PR to adopt some of my new shared infrastructure, the suppression of the diagnostics is working more consistently.

I do think we'll want to start diagnosing when an `@objc @implementation extension` is less available than the interface it's implementing. As it stands, this it allows a back door into calling arbitrary unavailable code which ought to be unreachable.